rosmon_core/src/diagnostics/rosmon_to_diagnostic.cpp

```diff
@@ -73,3 +75,24 @@ void RosmonToDiagnostic::updateDiagnostics(const std::vector<NodeMonitor::Ptr>&
     }
     m_diagnosticsPublisher.publish(currentDiagnosticArray);
 }
+
+std::string RosmonToDiagnostic::memoryToString(uint64_t memory)
+{
+    if(memory < 1000)
+    {
+        return fmt::format("{} B", memory);
+    }
+    else if(memory < 1e6)
+    {
+        return fmt::format("{:.2f} KiB", memory / 1e3);
```

Comment:
If we are talking `KiB`, you should check against 1024*1024 and divide by 1024, not 1000. Sorry, should have mentioned that.